cfpb/cfgov-refresh
# Add HTML editor and tip styling button for Ask CFPB Answers

Adds buttons to the RichTextField editor on Ask admin pages so content editors can continue to create tips/warnings in Ask CFPB answers. 

## Additions

- HTML editor in Ask admin RichTextFields
- Tip styling button for Ask admin RichTextFields
- CSS to style tips in RichTextFields on Ask admin pages
- CSS to style tips on Ask answer pages

## Testing

- Check out branch 
- Set up Ask CFPB per instructions in #2780 
- Run gulp
- Navigate to [an Ask CFPB question](http://localhost:8000/admin/ask_cfpb/answer/edit/2061/) in the admin
  - Click on the `Edit HTML` button on the Answer field to bring up the html editor. Verify that changes made to the HTML are preserved when you save the answer and [preview the associated question page](http://localhost:8000/admin/pages/6950/edit/)
  - Create a test tip header & paragraph in the Answer RichTextField. Highlight the tip and click the tip styling button. Verify that the tip is wrapped in an `<aside class="answer-module>` & appears with the correct styling in the RichTextField and on the associated question page.
- Navigate to a non-Ask page in the admin and verify that the tip and html editor buttons do not appear on RichTextFields

## Screenshots

#### HTML editing:

Editor button
<img width="693" alt="screen shot 2017-03-27 at 2 57 06 pm" src="https://cloud.githubusercontent.com/assets/778171/24379818/ab8f792a-12fd-11e7-91c5-94b843f1e0d8.png">  

  
Editor view
<img width="862" alt="screen shot 2017-03-27 at 2 56 36 pm" src="https://cloud.githubusercontent.com/assets/778171/24379829/adda5c9a-12fd-11e7-85f4-67fc20ed0747.png">
  

#### Tip button
<img width="684" alt="screen shot 2017-03-27 at 2 43 25 pm" src="https://cloud.githubusercontent.com/assets/778171/24379860/d875eeec-12fd-11e7-8b46-670b6469abfa.png">

  
<img width="688" alt="screen shot 2017-03-27 at 2 44 01 pm" src="https://cloud.githubusercontent.com/assets/778171/24379871/e1831c8a-12fd-11e7-8e3d-f5ffe2282f39.png">


## Notes

- Eventually we want to change the way tips are created so editors don't need to apply classes to them, but for now this replicates existing functionality
- We may only need one of the buttons; this PR allows content editors to assess them both on build

## Checklist
